analyses/cnv-chrom-plot/cn_status_heatmap.Rmd

```diff
@@ -380,51 +375,36 @@ This annotation object strategy was originally from [chromosomal-instability](ht
 histologies <-
   data.frame(Kids_First_Biospecimen_ID = bin_calls_df$biospecimen_id) %>%
   dplyr::inner_join(metadata %>%
-    dplyr::select(Kids_First_Biospecimen_ID, short_histology, sample_color)) %>%
+    dplyr::select(Kids_First_Biospecimen_ID, display_group, display_order, hex_codes)) %>%
   # Count numbers of samples per histology group and make new variable with counts
-  dplyr::group_by(short_histology) %>% 
-  dplyr::mutate(group_n = dplyr::n()) %>% 
-  # Ungroup the data
-  dplyr::ungroup() %>% 
-  # Temporarily we will put the n = 1 `short_histology` samples in the `Other` group. 
-  #TODO: Remove lines 387 - 400 when this is split into two panels
-  dplyr::mutate(
-    short_histology = dplyr::case_when(
-      group_n < min_group_size ~ "Other", 
-      TRUE ~ as.character(short_histology))) %>% 
-  # Reapply colors to groups
-   dplyr::mutate(sample_color = dplyr::recode(
-    short_histology,
-    !!!histology_col_palette
-  )) %>%
-  # ReCount numbers after the Other switches
-  dplyr::group_by(short_histology) %>% 
+  dplyr::group_by(display_group) %>% 
   dplyr::mutate(group_n = dplyr::n()) %>% 
   # Ungroup the data
   dplyr::ungroup() %>% 
   # Add sample sizes
-  dplyr::mutate(short_histology = factor(paste0(short_histology, " (n = ", group_n, ")"))) %>%
-  # Put in alphabetical order
-  dplyr::arrange(short_histology) %>%
-  # ComplexHeatmap wants this. 
-  tibble::column_to_rownames("Kids_First_Biospecimen_ID")
+  dplyr::mutate(display_group = factor(paste0(display_group, " (n = ", group_n, ")"))) %>% 
+  # Reorder display_group based on display_order
+  dplyr::mutate(display_group = forcats::fct_reorder(display_group, display_order)) %>%
+  # Make sure they are in display_order
+  dplyr::arrange(display_order) %>%
+  # Store as rownames
+  tibble::column_to_rownames("Kids_First_Biospecimen_ID") %>%
+  # We don't want this to actually be displayed on the heatmap though
+  dplyr::select(-display_order)
 ```
 
 Make a color key that's formatted for ComplexHeatmap. 
 
 ```{r}
 # Make color key specific to these samples
-histologies_color_key_filtered <- unique(histologies$sample_color)
-names(histologies_color_key_filtered) <- unique(histologies$short_histology)
-
-# Drop this column so ComplexHeatmap isn't tempted to plot it
-histologies <- dplyr::select(histologies, -sample_color, -group_n)
+histologies_color_key_filtered <- unique(histologies$hex_codes)
+names(histologies_color_key_filtered) <- unique(histologies$display_group)
```

Comment:
I'm a little worried about separate `unique()` calls preserving order... I think it should work fine, but it seems somehow dangerous to me. So I present an alternative below.

```suggestion
histologies_color_key_filtered <- histologies %>%
  dplyr::select(display_group, hex_codes) %>%
  dplyr::distinct() %>%
  dplyr::pull(hex_codes, name = display_group)
```

Reply:
Yes. I like this better.

Reply:
Turns out the name argument for `pull` is not in our version of dplyr for our docker image, so I can't use the last part, but I will add a distinct to this part but in my subsequent PR of this series.